### PR TITLE
[SPARK-46550][BUILD][SQL] Upgrade `datasketches-java` to 5.0.1

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to upgarde `datasketches-java` from 3.3.0 to 5.0.1.
Additionally, this PR adds handling for `SketchesArgumentException` to ensure backward compatibility for exceptional test scenarios in `hll.sql`.

### Why are the changes needed?
The new brings some bug fix and improvements, like 

- https://github.com/apache/datasketches-java/pull/437
- https://github.com/apache/datasketches-java/pull/482

The full release notes as follows:
- https://github.com/apache/datasketches-java/releases/tag/4.0.0
- https://github.com/apache/datasketches-java/releases/tag/4.1.0
- https://github.com/apache/datasketches-java/releases/tag/4.2.0
- https://github.com/apache/datasketches-java/releases/tag/5.0.0
- https://github.com/apache/datasketches-java/releases/tag/5.0.1

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/datasketchesAggregates.scala [hunk 1]
```diff
@@ -315,7 +316,7 @@ case class HllUnionAgg(
             union.update(sketch)
             Some(union)
           } catch {
-            case _: java.lang.Error =>
+            case _: SketchesArgumentException | _: java.lang.Error =>
```
Comment (LuciferYang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/datasketchesAggregates.scala:319]: The addition of capturing `SketchesArgumentException` is to fix the test failure for `hll.sql`.

```
diff --git a/sql/core/src/test/resources/sql-tests/results/hll.sql.out b/sql/core/src/test/resources/sql-tests/results/hll.sql.out
index adf6c40d1bd..8f4d14fcf8d 100644
--- a/sql/core/src/test/resources/sql-tests/results/hll.sql.out
+++ b/sql/core/src/test/resources/sql-tests/results/hll.sql.out
@@ -220,14 +220,8 @@ SELECT hll_sketch_estimate(CAST ('abc' AS BINARY))
 -- !query schema
 struct<>
 -- !query output
-org.apache.spark.SparkRuntimeException
-{
-  "errorClass" : "HLL_INVALID_INPUT_SKETCH_BUFFER",
-  "sqlState" : "22546",
-  "messageParameters" : {
-    "function" : "`hll_sketch_estimate`"
-  }
-}
+org.apache.datasketches.common.SketchesArgumentException
+Bounds Violation: reqOffset: 0, reqLength: 8, (reqOff + reqLen): 8, allocSize: 3
 
 
 -- !query
@@ -235,14 +229,8 @@ SELECT hll_union(CAST ('abc' AS BINARY), CAST ('def' AS BINARY))
 -- !query schema
 struct<>
 -- !query output
-org.apache.spark.SparkRuntimeException
-{
-  "errorClass" : "HLL_INVALID_INPUT_SKETCH_BUFFER",
-  "sqlState" : "22546",
-  "messageParameters" : {
-    "function" : "`hll_union`"
-  }
-}
+org.apache.datasketches.common.SketchesArgumentException
+Bounds Violation: reqOffset: 0, reqLength: 8, (reqOff + reqLen): 8, allocSize: 3
 
 
 -- !query
@@ -251,14 +239,8 @@ FROM (SELECT CAST('abc' AS BINARY) AS buffer)
 -- !query schema
 struct<>
 -- !query output
-org.apache.spark.SparkRuntimeException
-{
-  "errorClass" : "HLL_INVALID_INPUT_SKETCH_BUFFER",
-  "sqlState" : "22546",
-  "messageParameters" : {
-    "function" : "`hll_union_agg`"
-  }
-}
+org.apache.datasketches.common.SketchesArgumentException
+Bounds Violation: reqOffset: 0, reqLength: 8, (reqOff + reqLen): 8, allocSize: 3
```

**Comment by bjornjorgensen** [comment]
should we wait for 5.0.1 https://github.com/apache/datasketches-java/issues/484 
[it looks like the voting for it has passed.](https://lists.apache.org/thread/2odrgnt1wbsk6pkbfonr64v79lowpd85)

**Comment by LuciferYang** [comment]
@bjornjorgensen ok, let's wait 5.0.1. Thanks ~

**Comment by leerho** [comment]
5.01 fixed two issues: 
- PR 482: The Hll Union :: toString(), which prints out a simple diagnostic summary of the sketch, might change the internal state of the union.  This was not intended.  
- PR 485: The KllItemsSketch<Boolean> was not serializing and deserializing the min and max values properly. It only affects this specific generic case of <Boolean>. 
This is a rather bizarre use case for quantiles -- but nonetheless it is fixed!  :)

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun @leerho and @bjornjorgensen ~

**Pull request closed without merging**